File: MAP/LAB7-MAP/LAB7-MAP/Statements/PrgState.cs
using MAP_Lab7.Files;
using MAP_Lab7.Model;
using System.Text;

namespace MAP_Lab7.Statements
{
    public class PrgState
    {
        private IDictionaryy<string, int> dict;
        private IExeStack<Statement> stack;
        private IListt<int> outputList;
        private Statement stmt;
        private FileTable<int, FileData> fileTable;

        public PrgState(IDictionaryy<string, int> d, IExeStack<Statement> exes, IListt<int> ol, Statement s, FileTable<int, FileData> ft)
        {
            dict = d;
            stack = exes;
            outputList = ol;
            stmt = s;
            fileTable = ft;
        }

        public IDictionaryy<string, int> Dict
        {
            get { return dict; }
            set { dict = value; }
        }

        public Statement Stmt
        {
            get { return stmt; }
            set { stmt = value; }
        }

        public FileTable<int, FileData> FileTable
        {
            get { return fileTable; }
            set { fileTable = value; }
        }

        public IExeStack<Statement> ExeStack
        {
            get { return stack; }
            set { stack = value; }
        }

        public IListt<int> OutputList
        {
            get { return outputList; }
            set { outputList = value; }
        }

        public override string ToString()
        {
            StringBuilder buff = new StringBuilder();
            buff.Append(Dict);
            buff.Append(ExeStack);
            buff.Append(OutputList);
            buff.Append(Stmt);
            buff.Append("\n");
            return buff.ToString();
        }

    }
}

File: MAP/Lab6/src/Model/Statement/PrgState.java

package Model.Statement;

import Model.File.*;
import Model.*;
import Model.Heap.*;
import java.lang.StringBuffer;


public class PrgState
{
    private IExecStack<Statement> execStack;
    private IDictionary<String,Integer> symbolT;
    private IList<Integer> list;
    private FileTable<Integer,FileData> ft;
    private IHeap<Integer,Integer> heap;
    private Integer ID;

    public PrgState(IExecStack<Statement> s, IDictionary<String, Integer> d, IList<Integer> l, FileTable<Integer, FileData> ftable, IHeap<Integer, Integer> hp, Integer i)
    {
        execStack = s;
        symbolT = d;
        list = l;
        ft = ftable;
        heap = hp;
        ID = i;
    }

    public IExecStack<Statement> getExecStack()
    {
        return execStack;
    }

    public IDictionary<String, Integer> getSymbolT()
    {
        return symbolT;
    }

    public IList<Integer> getList()
    {
        return list;
    }

    public void setExecStack(IExecStack<Statement> execStack)
    {
        this.execStack = execStack;
    }

    public void setSymbolT(IDictionary<String, Integer> symbolT)
    {
        this.symbolT = symbolT;
    }

    public void setList(IList<Integer> l)
    {
        list = l;
    }

    public FileTable<Integer, FileData> getFileTable() { return ft; }

    public void setFileTable(FileTable<Integer, FileData> flt) { ft = flt; }

    public IHeap<Integer, Integer> getHeap() { return heap; }

    public Integer getID() { return ID; }

    public void setID(Integer i) { ID = i; }

    public boolean isNotCompleted()
    {
        return !execStack.isEmpty();
    }

    public PrgState oneStep()
    {
        if (execStack.isEmpty())
            throw new RuntimeException("Empty stack.");
        Statement statement = execStack.pop();
        return statement.execute(this);
    }

    @Override
    public String toString()
    {
        StringBuffer buff = new StringBuffer();
        buff.append(ID);
        buff.append(execStack);
        buff.append(symbolT);
        buff.append(list);
        buff.append(heap);
        buff.append("\n");
        return buff.toString();
    }
}